adClickFlow/shared/testCases.json

```diff
@@ -0,0 +1,494 @@
+              "url": "https://convert.ad-company.site/convert.js?ad=1",
+              "status": "loaded"
+            },
+            {
+              "url": "https://convert.ad-company.site/ping.gif",
+              "status": "loaded"
+            },
+            {
+              "url": "https://www.ad-company.site/track.js?ad=1",
+              "status": "blocked"
+            },
+            {
+              "url": "https://www.ad-company.site/ping.gif",
+              "status": "parent blocked"
+            }
+          ]
+        }
+      }
+    ]
+  },
+  {
+    "name": "y.js ad-company-provided; ad_domain provided (u3 not needed)",
+    "steps": [
+      {
+        "name": "Navigate to search page",
+        "action": {
+          "type": "navigate",
+          "url": "https://www.search-company.site/"
+        },
+        "expected": {
+          "url": "https://www.search-company.site/"
+        }
+      },
+      {
+        "name": "Click on [Ad 7]",
+        "action": {
+          "type": "click",
+          "id": "ad-id-7"
+        },
+        "expected": {
+          "url": "https://www.publisher-company.site/product.html?p=12",
+          "requests": [
+            {
+              "url": "https://convert.ad-company.site/convert.js?ad=1",
+              "status": "loaded"
+            },
+            {
+              "url": "https://convert.ad-company.site/ping.gif",
+              "status": "loaded"
+            },
+            {
+              "url": "https://www.ad-company.site/track.js?ad=1",
+              "status": "blocked"
+            },
+            {
+              "url": "https://www.ad-company.site/ping.gif",
+              "status": "parent blocked"
+            }
+          ]
+        }
+      }
+    ]
+  },
+  {
+    "name": "m.js ad-company-provided; ad_domain provided (dsl not needed)",
+    "steps": [
+      {
+        "name": "Navigate to search page",
+        "action": {
+          "type": "navigate",
+          "url": "https://www.search-company.site/"
+        },
+        "expected": {
+          "url": "https://www.search-company.site/"
+        }
+      },
+      {
+        "name": "Click on [Ad 8]",
+        "action": {
+          "type": "click",
+          "id": "ad-id-8"
+        },
+        "expected": {
+          "newTab": true,
+          "url": "https://www.publisher-company.site/product.html?p=200",
+          "requests": [
+            {
+              "url": "https://convert.ad-company.site/convert.js?ad=1",
+              "status": "loaded"
+            },
+            {
+              "url": "https://convert.ad-company.site/ping.gif",
+              "status": "loaded"
+            },
+            {
+              "url": "https://www.ad-company.site/track.js?ad=1",
+              "status": "blocked"
+            },
+            {
+              "url": "https://www.ad-company.site/ping.gif",
+              "status": "parent blocked"
+            }
+          ]
+        }
+      }
+    ]
+  },
```

Comment:
You could expand these test cases with Ads 9 - 14 in https://app.asana.com/0/72649045549333/1202521973600829/f

Reply:
Yeah that's the plan, I just wanted us to merge the current code then I can start on these.